.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,63 @@
+name: Unit tests
+
+on:
+  workflow_call:
+    inputs:
+        linux_5_8_enabled:
+            type: boolean
+            description: "Boolean to enable the Linux 5.8 Swift version matrix job. Defaults to true."
+            default: true
+        linux_5_8_arguments_override:
+            type: string
+            description: "The arguments passed to swift test in the Linux 5.8 Swift version matrix job."
+            default: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error"
+        linux_5_9_enabled:
+            type: boolean
+            description: "Boolean to enable the Linux 5.9 Swift version matrix job. Defaults to true."
+            default: true
+        linux_5_9_arguments_override:
+            type: string
+            description: "The arguments passed to swift test in the Linux 5.9 Swift version matrix job."
+            default: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error"
+        linux_5_10_enabled:
+            type: boolean
+            description: "Boolean to enable the Linux 5.10 Swift version matrix job. Defaults to true."
+            default: true
+        linux_5_10_arguments_override:
+            type: string
+            description: "The arguments passed to swift test in the Linux 5.10 Swift version matrix job."
+            default: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error"
+        linux_nightly_6_0_enabled:
+            type: boolean
+            description: "Boolean to enable the Linux nightly 6.0 Swift version matrix job. Defaults to true."
+            default: true
+        linux_nightly_6_0_arguments_override:
+            type: string
+            description: "The arguments passed to swift test in the Linux nightly 6.0 Swift version matrix job."
+            default: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error"
+        linux_nightly_main_enabled:
+            type: boolean
+            description: "Boolean to enable the Linux nightly main Swift version matrix job. Defaults to true."
+            default: true
+        linux_nightly_main_arguments_override:
+            type: string
+            description: "The arguments passed to swift test in the Linux nightly main Swift version matrix job."
+            default: "-Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error"
+ 
+jobs:
+    unit-tests:
+        name: Unit tests
+        uses: ./.github/workflows/pull_request_swift_matrix.yml
+        with:
+          name: "Unit tests"
+          matrix_linux_command: "swift test -Xswiftc -warnings-as-errors --explicit-target-dependency-import-check error"
```

Comment:
IIUC these flags shouldn't be here, since they are provided in the override in the other YAML file?

Wondering if the most acceptable thing to do would be for the default to command to just be `swift test` and allow folks to provide their override when they adopt, opting into whatever additional level of validation they want for their project.

Reply:
We need to pick a default here since this input is required. In practice this input will never actually be used because we provide an override for each version specific job. I do think we should have strong defaults though. In particular those two seem like net wins and I would rather people have to disable warnings as errors than the other way around.

Reply:
If you feel super strongly about it, then fine, but I would have had a slight lean the other way...

I think if we're offering this as a reusable building block it shouldn't be so opinionated so would prefer it for this to be free of anything that isn't the default, i.e. just `swift test`. I think that will be the least surprising behaviour. It would be annoying for folks to adopt the pipeline when their project currently builds fine (without additional strict flags, and then scratch their heads a little).

Reply:
You convinced me and I removed the default args now.